services/kinesis/src/it/java/software/amazon/awssdk/services/kinesis/SubscribeToShardIntegrationTest.java

```diff
@@ -114,61 +115,59 @@ public void subscribeToShard_ReceivesAllData() {
+        CountDownLatch countDownLatch = new CountDownLatch(2);
+        AtomicBoolean errorOccurred = new AtomicBoolean(false);
+        List<SubscribeToShardEventStream> events = new ArrayList<>();
+        asyncClient.subscribeToShard(r -> r.consumerARN(consumerArn)
+                                           .shardId(shardId)
+                                           .startingPosition(s -> s.type(ShardIteratorType.LATEST)),
+                                     new SubscribeToShardResponseHandler() {
+                                         @Override
+                                         public void responseReceived(SubscribeToShardResponse response) {
+                                             verifyHttpMetadata(response);
+                                         }
+
+                                         @Override
+                                         public void onEventStream(SdkPublisher<SubscribeToShardEventStream> publisher) {
+                                             publisher.limit(3).subscribe(new Subscriber<SubscribeToShardEventStream>() {
+                                                 @Override
+                                                 public void onSubscribe(Subscription subscription) {
+                                                     subscription.request(10);
+                                                 }
+
+                                                 @Override
+                                                 public void onNext(SubscribeToShardEventStream subscribeToShardEventStream) {
+                                                     events.add(subscribeToShardEventStream);
+                                                 }
+
+                                                 @Override
+                                                 public void onError(Throwable throwable) {
+                                                     errorOccurred.set(true);
+                                                 }
+
+                                                 @Override
+                                                 public void onComplete() {
+                                                     countDownLatch.countDown();
+                                                 }
+                                             });
+                                         }
+
+                                         @Override
+                                         public void exceptionOccurred(Throwable throwable) {
+                                             errorOccurred.set(true);
+                                         }
+
+                                         @Override
+                                         public void complete() {
+                                             countDownLatch.countDown();
+                                         }
+                                     }).join();
+
+        assertThat(countDownLatch.getCount()).isEqualTo(0);
```

Comment:
I don't think this is the correct behavior, count should still be 2. Cancelling the subscription should not call the `complete` since that should only be signaled when the full stream is read.